### PR TITLE
Feature/40 query ipfs gateways after add

### PR DESCRIPTION
**Pull request by 0x6431346e**

### CODE REVIEW

File: app/api.js [hunk 1]
```diff
@@ -36,7 +40,7 @@ export function initIPFSClient() {
 /**
  * This function will allow the user to add a file to the IPFS repo.
  */
-export function addFileFromFSPath(filePath) {
+export function addFileFromFSPath(filePath, _queryGateways = queryGateways) {
```
Comment (0x6431346e) [inline app/api.js]: I had to do some dependency injection here to be able to mock it

File: app/api.test.js [hunk 2]
```diff
@@ -70,8 +79,8 @@ describe('api.js', ()=>{
     })
   })
 
-  describe('unpinObject', ()=>{
-    it('should reject when IPFS is not started', ()=>{
+  describe('unpinObject', () => {
```
Comment (0x6431346e) [inline app/api.test.js]: I can see our editors like to format differently :smile: , I will look into this at some point

**Comment by 0x6431346e** [comment]
I've added the implementation for https://dev.siderus.team/issues/41 in this PR as well, hope that's alright

**Comment by koalalorenzo** [comment]
That is ok

File: app/api.js [hunk 3]
```diff
@@ -94,11 +99,28 @@ export function addFileFromFSPath(filePath) {
                    */
                   .then(res => resolve([...items, wrapper]))
               )
+            if (!Settings.getSync('skipGatewayQuery')) {
```
Comment (koalalorenzo) [inline app/api.js]: * Fix the indentation
* refactor the `.then` chain in callback style
* query also the links (not just the wrapper hash, as it does not contain the data)

**Comment by koalalorenzo** [comment]
LGTM, feel free to merge

**Pull request merged**